lib/_http_client.js

```diff
@@ -241,22 +241,7 @@ function ClientRequest(options, cb) {
     this._deferToConnect(null, null, () => this._flush());
   };
 
-  var newSocket;
-  if (this.socketPath) {
-    this._last = true;
-    this.shouldKeepAlive = false;
```

Comment:
@bnoordhuis I tried to do a deep blame, seems like it's your code https://github.com/nodejs/node/blob/bc7cfd7cd7c2e513a9ac9a2820f3b6f7331735b5/lib/http.js#L1025
Is there a specific reason why POSIX sockets `shouldKeepAlive = false`?